spec/rubocop/cop/rspec/sort_metadata_spec.rb

```diff
@@ -0,0 +1,206 @@
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      it 'Something', :a, :b, baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using mixed metadata ' \
+     'and both symbols metadata and hash keys are not in alphabetical order' do
+    expect_offense(<<~RUBY)
+      it 'Something', :b, :a, foo: 'bar', baz: true do
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      it 'Something', :a, :b, baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using mixed metadata ' \
+     'and both symbols metadata and hash keys are not in alphabetical order ' \
+     'and the hash values are complex objects' do
+    expect_offense(<<~RUBY)
+      it 'Something', :b, :a, foo: ->(x) { bar(x) }, baz: Snafu.new do
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      it 'Something', :a, :b, baz: Snafu.new, foo: ->(x) { bar(x) } do
+      end
+    RUBY
+  end
+
+  it 'does not register an offense ' \
+     'when the metadata is not on one single line' do
+    expect_no_offenses(<<~RUBY)
+      RSpec.describe 'Something', :foo, :bar,
+                                   baz: 'goo' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using only symbol metadata ' \
+     'in a config block, but not in alphabetical order' do
+    expect_offense(<<~RUBY)
+      RSpec.configure do |c|
+        c.before(:each, :b, :a) { freeze_time }
+                        ^^^^^^ Sort metadata alphabetically.
+        c.after(:each, foo: 'bar', baz: true) { travel_back }
+                       ^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      RSpec.configure do |c|
+        c.before(:each, :a, :b) { freeze_time }
+        c.after(:each, baz: true, foo: 'bar') { travel_back }
+      end
+    RUBY
+  end
+
+  context 'when using custom RSpec language ' \
+          'without adjusting the RuboCop RSpec language configuration' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        RSpec.configure do |c|
+          c.antes(:each, :y, :x) { freeze_time }
+        end
+
+        RSpec.describan "Algo", :b, :a do
+          contexto_compartido 'una situación complicada', foo: 'bar', baz: true do
+          end
+
+          ejemplo "hablando español", foo: 'bar', baz: true do
+          end
+        end
+      RUBY
+    end
+  end
+
+  context 'when using custom RSpec language ' \
+          'and adjusting the RuboCop RSpec language configuration' do
+    let(:config) do
+      RuboCop::ConfigLoader.merge_with_default(
+        RuboCop::Config.new(YAML.safe_load(language_config)), ''
+      )
+    end
+
+    let(:language_config) do
```

Comment:
We only check language config in one cop's specs:
```ruby
    before do
      other_cops.dig('RSpec', 'Language', 'Includes', 'Examples')
        .push('it_has_special_behavior')
    end
```
but I love your approach better 👍

Reply:
I really mean it.
But if you decided to go with `other_cops.dig.push` approach, this definition becomes redundant.